File: retroauth-core/src/test/java/com/andretietz/retroauth/testimpl/TestInterface.java
package com.andretietz.retroauth.testimpl;

import com.andretietz.retroauth.Authenticated;

import retrofit2.http.GET;
import rx.Observable;

/**
 * Created by andre on 11.06.2016.
 */

public interface TestInterface {

    int OWNER_TYPE = 1;
    int TOKEN_TYPE = 2;

    String TEST_BODY = "{ \"value\" : 1 }";

    @Authenticated({OWNER_TYPE, TOKEN_TYPE})
    @GET("some/authenticated/path")
    Observable<TestResponse> authenticatedMethod();

    @GET("some/unauthenticated/path")
    Observable<TestResponse> unauthenticatedMethod();
}

File: retroauth/src/test/java/com/andretietz/retroauth/testimpl/TestInterface.kt
package com.andretietz.retroauth.testimpl

import com.andretietz.retroauth.Authenticated
import io.reactivex.Single
import retrofit2.http.GET

interface TestInterface {

    companion object {
        const val TEST_BODY = "{ \"value\" : 1 }"
    }

    @Authenticated
    @GET("some/authenticated/path")
    fun authenticatedMethod(): Single<TestResponse>

    @GET("some/unauthenticated/path")
    fun unauthenticatedMethod(): Single<TestResponse>

}
